specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/jobs.json

```diff
@@ -0,0 +1,879 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Machine Learning ManagementFrontEnd Service",
+    "version": "2020-09-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/jobs/{id}": {
+      "put": {
```

Comment:
Does put starts the job also? What if job fails and needs to be restarted how will it work?
Have you think about the pattern of JOB creation independent of job operations like start/stop?

Reply:
yes, PUT starts job also, if job fails user can submit new job with same inputs.
for machine learning tracking lineage is really important so we dont allow update on most of inputs in jobs.
we do allow cancel job and delete job.
we have thought about independent job creation but feedback we got from user is they prefer starting job as part of create instead of one extra API call.
that said, that is why we are not publishing this APIs to public yet and hence doNotMerge tag so we can get early feedback and change accordingly.
we had detailed discussion with ARM team in Aug regarding this.